public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx

```diff
@@ -73,22 +73,14 @@ export function ChannelSubForm<R extends ChannelValues>({
   useEffect(() => {
     // Restore values when switching back from a changed integration to the default one
     const subscription = watch((v, { name, type }) => {
-      const value = name ? v[name] : '';
+      const value = name ? getValues(name) : '';
       if (initialValues && name === fieldName('type') && value === initialValues.type && type === 'change') {
         setValue(fieldName('settings'), initialValues.settings);
       }
-      // Restore initial value of an existing oncall integration
-      if (
-        initialValues &&
-        name === fieldName('settings.integration_type') &&
-        value === OnCallIntegrationType.ExistingIntegration
-      ) {
-        setValue(fieldName('settings.url'), initialValues.settings.url);
-      }
```

Comment:
I think we still need this. 
`integration_type` is a kind of artificial settings' property just for a nice UX (dropdown of Oncall integrations' names instead of URLs)
What we really store in the integration is the `url` of the OnCall integration